spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/ReleaseAnalyzer.java

```diff
@@ -168,7 +190,12 @@ private ReleaseAnalysisReport createReleaseAnalysisReport(Release existingReleas
 					StringUtils.collectionToCommaDelimitedString(releaseDifference.getChangedApplicationNames()) + "]");
 			appsToUpgrade.addAll(releaseDifference.getChangedApplicationNames());
 		}
-		return new ReleaseAnalysisReport(appsToUpgrade, releaseDifference, existingRelease, replacingRelease);
+		// Set explicit app names from the client only if the update is enforced.
+		else if (isForceUpdate && appNamesToUpdate != null) {
```

Comment:
If a stream update also passes in properties, then the force option is ignored.  I guess we should take into account that some apps in the stream maybe updated due the stream update changing properties in addition to some apps being listed in the force option.  Otherwise, we should make the command for stream update (from SCDF perspective) have mutually exclusive command arguments (properties and force).

Reply:
I played around with it, and it gets sort of complicated allowing both a `--propertiesFile` to be specified in addition to a `--force` option as well.  For example, in `ReleaseAnalysisReport` there is the code 

```
	public List<String> getApplicationNamesToUpgrade() {
		if (this.isForceUpdate) {
			return this.applicationNamesToUpgrade.size() > 0 ? this.applicationNamesToUpgrade : this.allApplicationNames;
		}
		else{
			return this.applicationNamesToUpgrade;
		}
	}
```

So given `http | tx | log`,  if we pass `--propertiesFile` for the `http` app and also `--force` then only `http` will be updated.  I think we need to introduce a new command `stream update-force` and corresponding new Controller that only accepts those options.  This makes the contract clear.

Reply:
I think I found a way to make the single command have a clear contract....

Reply:
I thought about this earlier. My thoughts were since the user wants to force upgrade a specific list of apps the properties correspond to other apps in the `--propertiesFile` should be ignored.